### PR TITLE
[Dev Tools] Add .devcontainer support

### PR DESCRIPTION
**Pull request by studna**
## Summary

Adds simple support for VSC Containers or Github Codespaces.

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [x] Dev Tools

## Testing

Open project in dev container using VSCode/Codespaces and run:
```sh
make go_develop_and_test
```

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code

### CODE REVIEW

**Comment by Olshansk** [comment]
@studna I've updated the metadata for this PR. Would appreciate if you could do so for future PRs please.

**Comment by Olshansk** [comment]
@okdas Can you PTAL at this PR?

I haven't reviewed it myself yet (on my backlog) but I know you've put some thought into this, so it will likely align/overlap with what you were planning.

**Review by okdas — APPROVED** [review]
That's a great start! We're going to need to install more dependencies for the full LocalNet to work if that's the intent. 
Thanks!

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
```
Comment (Olshansk) [inline .devcontainer/devcontainer.json:1]: Not a blocker, but there's an opportunity to update [1] so it's in the docs.

[1] https://github.com/pokt-network/poktroll/tree/main/docusaurus/docs/develop

**Review by Olshansk — APPROVED** [review]

**Comment by Olshansk** [comment]
@studna This is approved. Feel free to squash & merge when it's ready

**Comment by studna** [comment]
> That's a great start! We're going to need to install more dependencies for the full LocalNet to work if that's the intent. Thanks!

Thanks! Just to clarify, the intent is not to run the full LocalNet, just a possibility to develop inside of the container (or on Github Codespaces).

**Pull request merged**